modules/custom/lua/mixin_marid.lua

```diff
@@ -0,0 +1,151 @@
+------------------------------------
+-- Custom Module: Chigoes
+-- Use: Spawns Chigoes from set mobs
+------------------------------------
+require('modules/module_utils')
+------------------------------------
+local m = Module:new('chigoes')
+
+local mobs =
+{
+    { 'Wajaom_Woodlands',   'Marid'            },
+    { 'Wajaom_Woodlands',   'Grand_Marid'      },
+    { 'Bhaflau_Thickets',   'Marid'            },
+    { 'Bhaflau_Thickets',   'Grand_Marid'      },
+    { 'Aydeewa_Subterrane', 'Aydeewa_Diremite' },
+    { 'Caedarva_Mire',      'Wild_Karakul'     },
+    { 'Caedarva_Mire',      'Mosshorn'         },
+}
+
+-- validate that the file exists before trying to override it or weird things happen...
+local pathWithFilename = io.popen('cd'):read'*all'
+
+for _, entry in pairs(mobs) do
+    local zoneName = entry[1]
+    local mobName  = entry[2]
+    local filename = pathWithFilename:sub(1, -2) .. string.format('\\scripts\\zones\\%s\\mobs\\%s', zoneName, mobName) .. '.lua'
```

Comment:
There's absolutely no reason a module should have to hit the disk.  All of these scripts should be cached beforehand.

Reply:
some mobs don't have existing scripts. using xi.module.ensureTable on files that do exist has undesired results, i.e. some will stop working all together. I think fixing that is out of scope for this module, what do you propose?

Reply:
Everything is cached in the xi table.  It should be as simple as looking up there if something exists.